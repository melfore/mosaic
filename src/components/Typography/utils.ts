type VariantDefaultsType = {
  [key: string]: {
    bottomSpacing: boolean;
    truncated: boolean;
  };
};

const VARIANT_DEFAULTS: VariantDefaultsType = {
  body1: {
    bottomSpacing: false,
    truncated: false,
  },
  caption: {
    bottomSpacing: false,
    truncated: false,
  },
  h5: {
    bottomSpacing: true,
    truncated: true,
  },
  h6: {
    bottomSpacing: true,
    truncated: true,
  },
  overline: {
    bottomSpacing: false,
    truncated: false,
  },
  subtitle1: {
    bottomSpacing: false,
    truncated: false,
  },
};

export const getBottomSpacing = (variant: string, bottomSpacing?: boolean): boolean =>
  bottomSpacing === undefined ? VARIANT_DEFAULTS[variant].bottomSpacing : bottomSpacing;

export const getTruncate = (variant: string, truncated?: boolean): boolean =>
  truncated === undefined ? VARIANT_DEFAULTS[variant].truncated : truncated;

export const VARIANT_COMPONENT_MAP = {
  body1: "p",
  caption: "span",
  h5: "h1",
  h6: "h2",
  overline: "h4",
  subtitle1: "h3",
};
