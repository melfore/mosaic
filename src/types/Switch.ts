import { BaseType } from "./Base";

export enum SwitchSize {
  small = "small",
  medium = "medium", // default value
}

export interface SwitchType extends BaseType {
  value?: boolean;
  onChange?: (checked: boolean) => any | void;
  required?: boolean;
  size?: SwitchSize;
  disabled?: boolean;
}
