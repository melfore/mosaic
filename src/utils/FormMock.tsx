import React, { FC, ReactElement, useState } from "react";

interface FormMockType {
  onInputChange: Function;
  inputValue: null | number | string;
  onChangePropName?: string;
  valuePropName?: string;
}

const FormMock: FC<FormMockType> = ({
  children,
  onInputChange,
  inputValue,
  onChangePropName = "onChange",
  valuePropName = "value",
}) => {
  const [value, setValue] = useState(inputValue);
  const wrappedOnChange = (value: null | number | string) => {
    setValue(value);
    onInputChange(value);
  };

  return React.cloneElement(children as ReactElement<any>, {
    [onChangePropName]: wrappedOnChange,
    [valuePropName]: value,
  });
};

export default FormMock;
