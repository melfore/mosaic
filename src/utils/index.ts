import { ChangeEvent, MouseEvent } from "react";

export const suppressEvent = (event: ChangeEvent<any> | MouseEvent) => {
  event.preventDefault();
  event.stopPropagation();
};
