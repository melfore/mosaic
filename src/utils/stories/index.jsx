import React, { Fragment } from "react";
import CustomDocsPage from "./CustomDocsPage";

export const getDocsPageStructure = (name, blocks) => ({
  docs: {
    page: () => <CustomDocsPage blocks={blocks} name={name} />,
  },
});

export const StoriesWrapper = (props) => (
  <Fragment>
    <div className="stories-wrapper">{props.children}</div>
  </Fragment>
);
