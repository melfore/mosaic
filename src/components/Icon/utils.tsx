import React from "react";
import MUIAddIcon from "@material-ui/icons/Add";
import MUIDeleteIcon from "@material-ui/icons/Delete";
import MUIEditIcon from "@material-ui/icons/Edit";
import MUIPlaylistAddCheckIcon from "@material-ui/icons/PlaylistAddCheck";
import MUISendIcon from "@material-ui/icons/Send";
import { Icons } from "../../types/Icon";

export const iconsCatalog = {
  [Icons.add]: <MUIAddIcon />,
  [Icons.delete]: <MUIDeleteIcon />,
  [Icons.edit]: <MUIEditIcon />,
  [Icons.playlistAddCheck]: <MUIPlaylistAddCheckIcon />,
  [Icons.send]: <MUISendIcon />,
};
