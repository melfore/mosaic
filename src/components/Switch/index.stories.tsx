import React from "react";
import { action } from "@storybook/addon-actions";
import { boolean, text } from "@storybook/addon-knobs";
import { getDocsPageStructure, StoriesWrapper } from "../../utils/stories";
import Switch from ".";
import { SwitchSize } from "../../types/Switch";
import FormMock from "../../utils/mocks/FormMock";

export default {
  title: "Switch",
  component: Switch,
  parameters: {
    ...getDocsPageStructure("Switch", ["usage", "canvas"]),
  },
};

export const Canvas = () => (
  // FormMock simulates external form component handling state
  // In a real case scenario "onChange" and "value" props must be passed to Switch
  <FormMock inputValue={boolean("value", true)} onInputChange={action("Change switch")}>
    <Switch
      dataCy={text("data-cy", "switch-identifier")}
      value={boolean("value", true)}
      onChange={action("Change switch")}
    />
  </FormMock>
);

export const Values = () => (
  <StoriesWrapper>
    <Switch dataCy={"switch-identifier"} value={false} onChange={() => {}} />
    <Switch dataCy={"switch-identifier"} value={true} onChange={() => {}} />
  </StoriesWrapper>
);

export const Disabled = () => (
  <StoriesWrapper>
    <Switch dataCy={"switch-identifier"} value={false} disabled />
    <Switch dataCy={"switch-identifier"} value={true} disabled />
  </StoriesWrapper>
);

export const Size = () => (
  <StoriesWrapper>
    <Switch dataCy={"switch-identifier"} value={true} />
    <Switch dataCy={"switch-identifier"} value={true} size={SwitchSize.small} />
    <Switch dataCy={"switch-identifier"} value={false} />
    <Switch dataCy={"switch-identifier"} value={false} size={SwitchSize.small} />
  </StoriesWrapper>
);
