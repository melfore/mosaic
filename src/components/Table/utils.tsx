import React, { forwardRef, Fragment } from "react";
import { Action as MTActionType, Column as MTColumnType, Options as MTOptionsType } from "material-table";
import { Icons, IconSize } from "../../types/Icon";
import { TableActionScope, TableActionType, TableColumnType } from "../../types/Table";
import Button from "../Button";
import Icon from "../Icon";
import IconButton from "../IconButton";

// For default props refer to Options section here https://material-table.com/#/docs/all-props
export const DEFAULT_TABLE_OPTIONS: MTOptionsType = {
  debounceInterval: 500,
  draggable: false,
  emptyRowsWhenPaging: false,
};

export const actionAdapter = (action: TableActionType): MTActionType<object> => {
  const {
    callback,
    label,
    disabled = false,
    hidden = false,
    icon = undefined,
    scope = TableActionScope.default,
  } = action;
  return {
    disabled,
    hidden,
    icon: `${label}$${icon}`,
    isFreeAction: scope === TableActionScope.default,
    onClick: callback,
  };
};

export const actionComponentAdapter = (props: any) => {
  const { data, action } = props;
  const { disabled, hidden, icon, isFreeAction, onClick }: MTActionType<object> = action;
  if (hidden) {
    return null;
  }

  const labelAndIconName = icon as string;
  const label = labelAndIconName.split("$")[0];
  const iconName = labelAndIconName.split("$")[1] as Icons;
  return (
    <Fragment>
      {isFreeAction && (
        <Fragment>
          <span>&nbsp;&nbsp;&nbsp;&nbsp;</span>
          <Button
            disabled={disabled}
            icon={!iconName ? undefined : { name: iconName }}
            label={label}
            onClick={(event: any) => {
              onClick(event, data);
              event.preventDefault();
              event.stopPropagation();
            }}
          />
        </Fragment>
      )}
      {!isFreeAction && (
        <Fragment>
          <IconButton
            disabled={disabled}
            icon={iconName}
            onClick={(event: any) => {
              onClick(event, data);
              event.preventDefault();
              event.stopPropagation();
            }}
            size={IconSize.small}
          />
          <span>&nbsp;&nbsp;</span>
        </Fragment>
      )}
    </Fragment>
  );
};

export const columnAdapter = (column: TableColumnType): MTColumnType<object> => {
  const { label, path, render } = column;
  return {
    field: path,
    render,
    title: label,
  };
};

export const iconAdapter = (name: Icons, size = IconSize.default) =>
  forwardRef((props: any, ref: any) => {
    return <Icon forwarded={{ ...props, ref }} name={name} size={size} />;
  });
