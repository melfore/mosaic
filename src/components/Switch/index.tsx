import React, { FC } from "react";
import MUISwitch from "@material-ui/core/Switch";
import { SwitchType, SwitchSize } from "../../types/Switch";

/**
 * Switch component made on top of `@material-ui/core/Switch
 */
const Switch: FC<SwitchType> = ({
  dataCy,
  value = false,
  onChange,
  required = false,
  size = SwitchSize.default,
  disabled = false,
}) => {
  const onChangeHandler = (event: any) => {
    const value = event.target.checked;
    if (onChange) {
      onChange(value);
    }
  };

  return (
    <MUISwitch
      color="primary"
      data-cy={dataCy}
      checked={value}
      onChange={onChangeHandler}
      required={required}
      size={size}
      disabled={disabled}
    />
  );
};

export default Switch;
