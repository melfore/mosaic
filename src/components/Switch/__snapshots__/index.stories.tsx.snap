// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inputs/Switch DisableRipple smoke-test 1`] = `
<label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-mui-classname"
       data-cy="switch"
>
  <span class="MuiSwitch-root MuiSwitch-sizeMedium css-mui-classname">
    <span class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-mui-classname"
          data-cy="switch-toggle"
    >
      <input class="PrivateSwitchBase-input MuiSwitch-input css-mui-classname"
             type="checkbox"
             data-cy="switch-input"
             checked
      >
      <span class="MuiSwitch-thumb css-mui-classname">
      </span>
    </span>
    <span class="MuiSwitch-track css-mui-classname">
    </span>
  </span>
</label>
`;

exports[`Inputs/Switch Disabled smoke-test 1`] = `
<label class="MuiFormControlLabel-root Mui-disabled MuiFormControlLabel-labelPlacementStart css-mui-classname"
       data-cy="switch"
>
  <span class="MuiSwitch-root MuiSwitch-sizeMedium css-mui-classname">
    <span class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-disabled Mui-disabled PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-disabled Mui-checked Mui-disabled css-mui-classname"
          tabindex="-1"
          aria-disabled="true"
          data-cy="switch-toggle"
    >
      <input class="PrivateSwitchBase-input MuiSwitch-input css-mui-classname"
             disabled
             type="checkbox"
             data-cy="switch-input"
             checked
      >
      <span class="MuiSwitch-thumb css-mui-classname">
      </span>
    </span>
    <span class="MuiSwitch-track css-mui-classname">
    </span>
  </span>
</label>
`;

exports[`Inputs/Switch Label smoke-test 1`] = `
<label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-mui-classname"
       data-cy="switch"
>
  <span class="MuiSwitch-root MuiSwitch-sizeMedium css-mui-classname">
    <span class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-mui-classname"
          data-cy="switch-toggle"
    >
      <input class="PrivateSwitchBase-input MuiSwitch-input css-mui-classname"
             type="checkbox"
             data-cy="switch-input"
             checked
      >
      <span class="MuiSwitch-thumb css-mui-classname">
      </span>
    </span>
    <span class="MuiSwitch-track css-mui-classname">
    </span>
  </span>
  <span class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-mui-classname">
    Switch
  </span>
</label>
`;

exports[`Inputs/Switch LabelEnd smoke-test 1`] = `
<label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-mui-classname"
       data-cy="switch"
>
  <span class="MuiSwitch-root MuiSwitch-sizeMedium css-mui-classname">
    <span class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-mui-classname"
          data-cy="switch-toggle"
    >
      <input class="PrivateSwitchBase-input MuiSwitch-input css-mui-classname"
             type="checkbox"
             data-cy="switch-input"
             checked
      >
      <span class="MuiSwitch-thumb css-mui-classname">
      </span>
    </span>
    <span class="MuiSwitch-track css-mui-classname">
    </span>
  </span>
  <span class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-mui-classname">
    Switch
  </span>
</label>
`;

exports[`Inputs/Switch Localized smoke-test 1`] = `
<label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-mui-classname"
       data-cy="locale.switch"
>
  <span class="MuiSwitch-root MuiSwitch-sizeMedium css-mui-classname">
    <span class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-mui-classname"
          data-cy="locale.switch-toggle"
    >
      <input class="PrivateSwitchBase-input MuiSwitch-input css-mui-classname"
             type="checkbox"
             data-cy="locale.switch-input"
             checked
      >
      <span class="MuiSwitch-thumb css-mui-classname">
      </span>
    </span>
    <span class="MuiSwitch-track css-mui-classname">
    </span>
  </span>
  <span class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-mui-classname">
    Interruttore
  </span>
</label>
`;

exports[`Inputs/Switch Primary smoke-test 1`] = `
<label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-mui-classname"
       data-cy="switch"
>
  <span class="MuiSwitch-root MuiSwitch-sizeMedium css-mui-classname">
    <span class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-mui-classname"
          data-cy="switch-toggle"
    >
      <input class="PrivateSwitchBase-input MuiSwitch-input css-mui-classname"
             type="checkbox"
             data-cy="switch-input"
             checked
      >
      <span class="MuiSwitch-thumb css-mui-classname">
      </span>
    </span>
    <span class="MuiSwitch-track css-mui-classname">
    </span>
  </span>
</label>
`;

exports[`Inputs/Switch SizeSmall smoke-test 1`] = `
<label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-mui-classname"
       data-cy="switch"
>
  <span class="MuiSwitch-root MuiSwitch-sizeSmall css-mui-classname">
    <span class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-mui-classname"
          data-cy="switch-toggle"
    >
      <input class="PrivateSwitchBase-input MuiSwitch-input css-mui-classname"
             type="checkbox"
             data-cy="switch-input"
             checked
      >
      <span class="MuiSwitch-thumb css-mui-classname">
      </span>
    </span>
    <span class="MuiSwitch-track css-mui-classname">
    </span>
  </span>
</label>
`;

exports[`Inputs/Switch Styled smoke-test 1`] = `
<label class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-mui-classname"
       data-cy="switch"
>
  <span class="MuiSwitch-root MuiSwitch-sizeMedium css-mui-classname">
    <span class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-mui-classname"
          data-cy="switch-toggle"
          style="color: red;"
    >
      <input class="PrivateSwitchBase-input MuiSwitch-input css-mui-classname"
             type="checkbox"
             data-cy="switch-input"
             checked
      >
      <span class="MuiSwitch-thumb css-mui-classname">
      </span>
    </span>
    <span class="MuiSwitch-track css-mui-classname">
    </span>
  </span>
</label>
`;
